app/plugin_install.go

```diff
@@ -290,6 +312,42 @@ func (a *App) removePluginLocally(id string) *model.AppError {
 	return nil
 }
 
+func (a *App) saveSignatures(pluginId string, signatures []string) *model.AppError {
+	for counter, signature := range signatures {
+		signatureBytes, err := base64.StdEncoding.DecodeString(signature)
+		if err != nil {
+			return model.NewAppError("saveSignatures", "app.plugin.signature_decode.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+
+		filePath := a.getSignatureStorePath(pluginId, counter+1)
```

Comment:
Sorry I know I had originally suggested to let the counter start at `1`. But thinking if there is any harm in starting the counter at `0`? That way we don't have to have this `counter+1` everywhere we assume index counters. I.e. we start from `0`.